### ISSUE TITLE
proc: leaky process when failing to debug stripped binaries in headless mode

### ISSUE THREAD

**Issue by polinasok** [go-delve/delve#2924]
1. What version of Delve are you using (`dlv version`)?
```
Delve Debugger
Version: 1.8.2
```
2. What version of Go are you using? (`go version`)?
```
go version go1.17.8 darwin/amd64
```
3. What operating system and processor architecture are you using?
```
MacOS Monterey
Version 12.2.1
```
4. What did you do?

Ran dlv dap debugging session with the following in the launch configuration
```
"buildFlags": "--ldflags '-s -w'"
```
5. What did you expect to see?
Launch error only

6.  What did you see instead?
Launch error and process printing "hello world"
```
2022-03-15T17:04:47-07:00 debug layer=dap building from "/Users/polina/delve": [go build -o /Users/polina/delve/__debug_bin -gcflags all=-N -l --ldflags -s -w /Users/polina/go/src/hello/hello.go]
debugserver-@(#)PROGRAM:LLDB  PROJECT:lldb-1205.0.27
 for x86_64.
Got a connection, launched process /Users/polina/delve/__debug_bin (pid = 83621).
Exiting.
>>>>>>>>>> hello world <<<<<<<<<<<
2022-03-15T17:04:47-07:00 debug layer=dap Failed to launch: could not launch process: decoding dwarf section info at offset 0x0: too short
2022-03-15T17:04:47-07:00 debug layer=dap [-> to client]{"seq":0,"type":"response","request_seq":2,"success":false,"command":"launch","message":"Failed to launch","body":{"error":{"id":3000,"format":"Failed to launch: could not launch process: decoding dwarf section info at offset 0x0: too short","showUser":true}}}
```

Looks like this was fixed for dlv-regular in #2018 under issue #2017.
```
$ dlv debug hello.go --build-flags="--ldflags '-s -w'"
could not launch process: decoding dwarf section info at offset 0x0: too short
$
```

**Comment by aarzilli** [comment]
Someone with convenient macOS access should experiment around with this. The problem is in pkg/proc/gdbserial/gdbserver.go, `initialize` function. When we can't proceed we close the connection to debugserver by calling `p.conn.conn.Close()`, if closing like that doesn't lead to debugserver properly shutting down maybe that's a debugserver bug, but maybe we can do something to make it better. Maybe `p.conn.detach()` works better.

**Comment by polinasok** [comment]
Looks like this is not a dap issue. I am able to reproduce the problem with headless dlv as well.
```
$ dlv debug _fixtures/increment.go --build-flags="--ldflags '-w'" --headless
API server listening at: 127.0.0.1:60943
debugserver-@(#)PROGRAM:LLDB  PROJECT:lldb-1205.0.27
 for x86_64.
Got a connection, launched process /Users/polina/delve/__debug_bin (pid = 6352).
Exiting.
4
could not launch process: decoding dwarf section info at offset 0x0: too short
```
